src/map/mob.cpp

```diff
@@ -4170,248 +4180,1014 @@ static void item_dropratio_adjust(t_itemid nameid, int mob_id, int *rate_adjust)
+		//	this->invalidWarning(node["AI"], "Invalid monster AI %s, defaulting to MONSTER_TYPE_06.\n", ai.c_str());
+		//	constant = MONSTER_TYPE_06;
+		//}
+
+		mob->status.mode = static_cast<e_mode>(constant);
+	} else {
+		if (!exists)
+			mob->status.mode = static_cast<e_mode>(MONSTER_TYPE_01);
+	}
+
+	if (this->nodeExists(node, "Class")) {
+		std::string class_;
+
+		if (!this->asString(node, "Class", class_))
+			return 0;
+
+		std::string class_constant = "CLASS_" + class_;
+		int64 constant;
+
+		if (!script_get_constant(class_constant.c_str(), &constant)) {
+			this->invalidWarning(node["Class"], "Unknown monster class %s, defaulting to CLASS_NORMAL.\n", class_.c_str());
+			constant = CLASS_NORMAL;
+		}
+
+		if (constant < CLASS_NORMAL || constant == 3 || constant > CLASS_EVENT) {
+			this->invalidWarning(node["Class"], "Invalid monster class %s, defaulting to CLASS_NORMAL.\n", class_.c_str());
+			constant = CLASS_NORMAL;
+		}
+
+		// Reset the modes on import
+		if (mob->status.class_ != CLASS_NORMAL)
+			mob->status.mode = static_cast<e_mode>(mob->status.mode & ~(MD_DETECTOR | MD_STATUSIMMUNE | MD_KNOCKBACKIMMUNE | MD_SKILLIMMUNE | MD_FIXEDITEMDROP));
+
+		mob->status.class_ = static_cast<uint8>(constant);
+
+		switch (constant) {
+			case CLASS_BOSS:
+				mob->status.mode = static_cast<e_mode>(mob->status.mode | (MD_DETECTOR | MD_STATUSIMMUNE | MD_KNOCKBACKIMMUNE));
+				break;
+			case CLASS_GUARDIAN:
+				mob->status.mode = static_cast<e_mode>(mob->status.mode | MD_STATUSIMMUNE);
+				break;
+			case CLASS_BATTLEFIELD:
+				mob->status.mode = static_cast<e_mode>(mob->status.mode | (MD_STATUSIMMUNE | MD_SKILLIMMUNE));
+				break;
+			case CLASS_EVENT:
+				mob->status.mode = static_cast<e_mode>(mob->status.mode | MD_FIXEDITEMDROP);
+				break;
+		}
+	} else {
+		if (!exists)
+			mob->status.class_ = CLASS_NORMAL;
+	}
+
+	if (this->nodeExists(node, "Attribute")) {
+		uint16 attr;
+
+		if (!this->asUInt16(node, "Attribute", attr))
+			return 0;
+
+		if (attr < 0 || attr > 127) {
+			this->invalidWarning(node["Attribute"], "Invalid monster attribute %d, skipping.\n", attr);
+			return 0;
 		}
 
-		// Delete the item
-		entry.mvpitem[i].nameid = 0;
-		entry.mvpitem[i].rate = 0;
+		// Reset the modes on import
+		mob->status.mode = static_cast<e_mode>(mob->status.mode & ~(MD_IGNOREMELEE | MD_IGNOREMAGIC | MD_IGNORERANGED | MD_MVP | MD_IGNOREMISC | MD_KNOCKBACKIMMUNE | MD_TELEPORTBLOCK));
+
+		if (attr > 0) {
+			if (attr & 1)
+				mob->status.mode = static_cast<e_mode>(mob->status.mode | MD_IGNOREMELEE);
+			if (attr & 2)
+				mob->status.mode = static_cast<e_mode>(mob->status.mode | MD_IGNOREMAGIC);
+			if (attr & 4)
+				mob->status.mode = static_cast<e_mode>(mob->status.mode | MD_IGNORERANGED);
+			if (attr & 8)
+				mob->status.mode = static_cast<e_mode>(mob->status.mode | MD_MVP);
+			if (attr & 16)
+				mob->status.mode = static_cast<e_mode>(mob->status.mode | MD_IGNOREMISC);
+			if (attr & 32)
```

Comment:
Should this add CLASS_BOSS condition to insert back MD_KNOCKBACKIMMUNE mode ?

because monster with Class: Boss and didn't have Attribute: 32  (ex. Baphomet) will lose MD_KNOCKBACKIMMUNE

Reply:
This attribute type is specifically knockback only seen from the documentation: `32: 0x200000 (cannot be knocked back)`.

Assigning `Boss` to the class field will apply knockback automatically: `01: 0x6200000 (boss class, immune to status changes, immune to knockback, detector)`

Reply:
I mean in src part

`mob->status.mode = static_cast<e_mode>(mob->status.mode & ~(MD_IGNOREMELEE | MD_IGNOREMAGIC | MD_IGNORERANGED | MD_MVP | MD_IGNOREMISC | MD_KNOCKBACKIMMUNE | MD_TELEPORTBLOCK));` 

inside  Attribute node will rip off MD_Knockbackimmune  

and never add back this mode if that monster is not have Attribute &32. 

So,  mob->status.mode will not have MD_KNOCKBACKIMMUNE despite   mob->status.mode == CLASS_BOSS

Reply:
this will happen if you did this in mob_db.yml

`Class: Boss
Attribute: (any without 32 bitmask)`